src/StructuredLogViewer/MainWindow.xaml.cs

```diff
@@ -659,31 +663,119 @@ private void Reload()
             OpenLogFile(logFilePath);
         }
 
-        private void SaveAs()
+        private async void RedactSecrets()

```

Comment:
async void is sketchy for various reasons :)

I usually use a pattern like this:

```
void Foo()
{
    FooAsync().Ignore(); // Ignore 
    _ = FooAsync(); // or like this, "observes" the task
}

async Task FooAsync()
{
}

        /// <summary>
        /// Consumes a task and doesn't do anything with it.  Useful for fire-and-forget calls to async methods within sync methods.
        /// </summary>
        /// <param name="task">The task whose result is to be ignored.</param>
        public static void Ignore(this Task task)
        {
        }
```